src/main/java/io/jenkins/plugins/appcenter/AppCenterRecorder.java

```diff
@@ -131,6 +139,96 @@ private boolean uploadToAppCenter(Run<?, ?> run, FilePath filePath, TaskListener
         }
     }
 
+    private String fetchReleaseNotes(Run<?, ?> build, FilePath filePath, EnvVars vars) {
+        File tempDir;
+        try {
+            tempDir = File.createTempFile("jtf", null);
+            if (tempDir.delete() && tempDir.mkdirs()) {
+                if (releaseNotesMethod instanceof ManualReleaseNotes) {
+                    ManualReleaseNotes manualReleaseNotes = (ManualReleaseNotes) releaseNotesMethod;
+                    if (((ManualReleaseNotes) releaseNotesMethod).getReleaseNotes() != null) {
+                        return manualReleaseNotes.getReleaseNotes();
+                    }
+                } else if (releaseNotesMethod instanceof FileReleaseNotes) {
+                    FileReleaseNotes fileReleaseNotes = (FileReleaseNotes) releaseNotesMethod;
```

Comment:
I would also break up this function into smaller ones. Ones that specifically handle the appropriate release notes methods. In fact you could move that logic to the Release notes methods class and return the output whenever you call `getReleaseNotes()`.